Insurgency/insserver

```diff
@@ -3,46 +3,87 @@
 # Server Management Script
 # Author: Daniel Gibbs
 # Website: http://gameservermanagers.com
-version="291015"
+version="031115"
 
 #### Variables ####
 
-# Notification Email
-# (on|off)
-emailnotification="off"
-email="email@example.com"
+# The name of this script file, used to show the LGSM link properly
+selfname=$(basename $(readlink -f "${BASH_SOURCE[0]}"))
 
-# Steam login
-steamuser="anonymous"
-steampass=""
+# Directories
+rootdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+scriptcfgdir="${rootdir}/cfg/servers"
+
+# Settings to get before config parsing begins
+engine="source"
+game="insurgency"
+gamename="Insurgency"
+servicename="$(basename $0)"
+
+#ipaddr=$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
 
-# Start Variables
+# Config files
+cfg_default="${scriptcfgdir}/_default.cfg"
+cfg_common="${scriptcfgdir}/_common.cfg"
+cfg_instance="${scriptcfgdir}/${servicename}.cfg"
+
+#If config directory does not exist, create it
+if [ ! -e $scriptcfgdir ]; then mkdir -p "$scriptcfgdir"; fi
+
+# If default config does not exist, create it. This should come from Git, and will be overwritten by updates.
+# Rather than try to wget it from Github or other fancy ways to get it, the simplest way to ensure it works is to simply create it here.
+fn_create_default_config(){
+cat <<'EOF' >> $cfg_default
+# Default config - Changes will be overwritten by updates.
+# Your settings for all servers go in _common.cfg
+# Server-specific settings go into $SERVER.cfg
+appid="237410"
+beta="" #To enable beta, use " -beta beta"
+clientport="27005"
 defaultmap="ministry"
+defaultmode="checkpoint"
+email="email@example.com"
+emailnotification="off" # (on|off)
+ip="0.0.0.0"
+lgsm_version="031115"
+logdays="7"
+mapcyclefile="mapcycle.txt"
 maxplayers="16"
+playlist="custom"
 port="27015"
 sourcetvport="27020"
-clientport="27005"
-ip="0.0.0.0"
+srcds_parms=""
```

Comment:
Thanks, I had that as an addon, but I just realized I should have put the reference to that variable in fn_parms after the last -parm and before the first +parm so that it executes in proper order. Or does srcds just know how to sort that out itself?